docs/std-encoding.md

```diff
@@ -25,3 +25,38 @@ Usage:
 | example | equals |
 |:-|:-|
 | `//encoding.json.decode('{"hi": "abc", "hello": 123}')` | `{'hello': 123, 'hi': (s: 'abc')}` |
+
+## `//encoding.proto.decode(proto <: bytes, rootModule <: string, message <: protocol buffers message) <: tuple`
+
+This method accepts [protocol buffers message](https://github.com/protocolbuffers/protobuf) information and data, and transfroms to a built-in arrai value.
+
+Sample code for [Sysl](https://github.com/anz-bank/sysl) protocol buffers message converting to arrai value:
+
+```arrai
+let sysl = //encoding.proto.decode(//encoding.proto.proto, //os.file('../translate/pb/test/sysl.pb'));
+let decodeSyslPb = //encoding.proto.decode(sysl);
+let shop = decodeSyslPb('Module', //os.file('../translate/pb/test/petshop.pb'));
+shop.apps('PetShopApi').attrs('package').s
+```
+
+It will output
+
+```arrai
+'io.sysl.demo.petshop.api'
+```
+
+Currently, it has to follow the code above to transfrom protocol buffers message to a built-in arrai value.
+
+```arrai
+let sysl = //encoding.proto.decode(//encoding.proto.proto, //os.file('../translate/pb/test/sysl.pb'));
+```
+
+In this code line, `//encoding.proto.proto` is a constant, `//os.file('../translate/pb/test/sysl.pb')` is binary file of protocol buffers message definition file `.proto`.
+
+```arrai
+let shop = decodeSyslPb('Module', //os.file("../translate/pb/test/petshop.pb"));
+```
+
+In this code line, `'Module'` is the root message type it want to start building arrai value from, `//os.file('../translate/pb/test/petshop.pb')` is binary file of protocol buffers message which is used as data source to build arrai value.
```

Comment:
You don't/shouldn't need to repeat the sample code you just showed. Cleaned up a bit:

```suggestion
The first line constructs a protobuf file descriptor. `//encoding.proto.proto` is a constant, requesting a file descriptor as output. `//os.file('translate/pb/test/sysl.pb')` is the binary output of compiling `sysl.proto` with `protoc`.

The second line uses the `sysl` file descriptor to parse `//os.file('translate/pb/test/petshop.pb')`, a compiled Sysl `Module` message.

The output is `shop`, a tuple representing a `Module`. It contains a field `apps`, which maps names to tuple representations of `Application` (and so on).```